01-fundamentos/secao-06-html-css-forms-flexbox-responsivo/dia-01-html-css-forms/script.js

```diff
@@ -0,0 +1,17 @@
+let btnSubmit = document.getElementById('submit');
+btnSubmit.addEventListener('click', function (event) {
+  event.preventDefault();
+  let checkboxPicture = document.getElementById('my-pictures').checked;
+  if (checkboxPicture == false) {
+    alert('Autorize o uso das imagens antes de enviar seus dados.')
+  }
+  let inputFullName = document.getElementById('full-name').value.length;
+  let inputEmail = document.getElementById('e-mail').value.length;
+  let textAreaReason = document.getElementById('longText').value.length;
+  if (inputFullName < 10 || inputFullName > 40 ||
+    inputEmail < 10 || inputEmail > 50 || textAreaReason > 500) {
+    alert('Dados Inválidos')
+  } else {
+    alert('Dados enviados com sucesso! Obrigado por participar do concurso TrybeTrip')
+  }
```

Comment:
Ao fazer validação de dados, seguir o fluxo de navegação do usuário, para exibir as mensagens de erro da mesma ordem que o usuário insere seus dados. Isso ajuda o usuário a entender, e identificar de forma mais rápida os campos que estão inválidos.

```suggestion
  // primeiro validar os inputs de texto
  let inputFullName = document.getElementById("full-name").value.length;
  let inputEmail = document.getElementById("e-mail").value.length;
  let textAreaReason = document.getElementById("longText").value.length;
  if (
    inputFullName < 10 ||
    inputFullName > 40 ||
    inputEmail < 10 ||
    inputEmail > 50 ||
    textAreaReason > 500
  ) {
    return alert("Dados Inválidos");
  }
  // depois validar o checkbox
  let checkboxPicture = document.getElementById("my-pictures").checked;
  if (checkboxPicture == false) {
    return alert("Autorize o uso das imagens antes de enviar seus dados.");
  }
  // as verificações passaram, exibir mensagem de sucesso
  return alert(
    "Dados enviados com sucesso! Obrigado por participar do concurso TrybeTrip"
  );
```